sycl/doc/GetStartedGuide.md

```diff
@@ -639,7 +639,7 @@ which contains all the symbols required.
 
 ## C++ standard
 
-* DPC++ runtime and headers require C++14 at least.
+* DPC++ runtime and headers require C++17 at least.
```

Comment:
@vladimirlaz, could you confirm that there are no runtime library ABI breaking changes requiring us to increment the library version, please?

Reply:
Idea for a separate PR: It would be good to have a static_assert somewhere in SYCL headers that would check that C++ is 17 or higher.
It would give user-friendly message instead of some unexpected compilation errors due to various reasons (using 'if constexpr', auto, inline vars, etc. with C++14 option)